Snowflake.Data/Core/SFStatement.cs

```diff
@@ -3,20 +3,107 @@
  */
 
 using System;
-using System.Web;
-using Newtonsoft.Json.Linq;
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using Snowflake.Data.Client;
-using Snowflake.Data.Core.FileTransfer;
 using Snowflake.Data.Log;
 using System.Threading;
 using System.Threading.Tasks;
 using System.Text;
+using System.ComponentModel;
 
 namespace Snowflake.Data.Core
 {
+    /// <summary>
+    /// The status types of the query.
+    /// </summary>
+    public enum QueryStatus
+    {
+        [StringAttr(value = "UNKNOWN")]
+        Unknown,
+        [StringAttr(value = "NO_DATA")]
+        NoData,
+        [StringAttr(value = "RUNNING")]
+        Running,
+        [StringAttr(value = "ABORTING")]
+        Aborting,
+        [StringAttr(value = "SUCCESS")]
+        Success,
+        [StringAttr(value = "FAILED_WITH_ERROR")]
+        FailedWithError,
+        [StringAttr(value = "ABORTED")]
+        Aborted,
+        [StringAttr(value = "QUEUED")]
+        Queued,
+        [StringAttr(value = "FAILED_WITH_INCIDENT")]
+        FailedWithIncident,
+        [StringAttr(value = "DISCONNECTED")]
+        Disconnected,
+        [StringAttr(value = "RESUMING_WAREHOUSE")]
+        ResumingWarehouse,
+        // purposeful typo
+        [StringAttr(value = "QUEUED_REPARING_WAREHOUSE")]
+        QueuedReparingWarehouse,
+        [StringAttr(value = "RESTARTED")]
+        Restarted,
+        [StringAttr(value = "BLOCKED")]
+        Blocked,
+    }
+
+    class StringAttr : Attribute
+    {
+        public string value { get; set; }
+    }
+
+    internal static class QueryStatusExtensions
+    {
+        internal static QueryStatus GetQueryStatusByStringValue(string description)

```

Comment:
rename description to value or stringValue

Reply:
Renamed